extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java

```diff
@@ -61,16 +69,44 @@ public void init() {
             }
 
             if (invoker.hasInput()) {
-                JavaType javaInputType = objectMapper.constructType(invoker.getInputType());
+                Type inputType = invoker.getInputType();
+
+                if (CloudEvent.class.equals(Reflections.getRawType(inputType))) {
+                    if (inputType instanceof ParameterizedType) {
+                        Type[] params = ((ParameterizedType) inputType).getActualTypeArguments();
+                        if (params.length == 1) {
+                            inputType = params[0];
+                            invoker.getBindingContext().put(INPUT_CE_DATA_TYPE, inputType);
+                        }
+                    } else {
+                        throw new RuntimeException("When using CloudEvent<> generic parameter must be used.");
```

Comment:
@mswiderski note that at least for now I am enforcing usage of the type parameter.

Reply:
type parameter should not be enforced and should default to what the current Funqy docs describe for defaults (sorry for repeating myself).